### DISCUSSION TITLE
Set default output path for vsh scripts

### DISCUSSION THREAD

**Discussion by Brixy** [vlang/v#21750] [Questions and Answers] [answered]
Hi!

I am considering V as our default scripting language (along with bash). Tests are very promising!

One thing bugs me: when you execute a `.vsh` script, the binary is compiled into the same directory. This is not convenient for scripting, as we track our scripts with git. So I would like to set a different output path for all `.vsh` scripts (but not for regular `.v` scripts), maybe either with a folder or project wide respected config file or an environment variable.

Is this possible?

Currently we use this workaround:

```
#!/usr/bin/env -S v -o ${HOME}/.local/share/v/my-script
```

Thank you very much.

**Comment by medvednikov** [comment]
Hi,

You can use the `VFLAGS` env var for this.

**Comment by Brixy** [comment]
Thanks a lot, and sorry for asking back.

It seems, that V does not allow paths without filenames with `-o`? Is this even a bug?

**Go:**

```
go build -o bin/ my-script.go
```

works. Directory is optinally created.

**V:**

```
v -o bin/ my-script.v
```

> builder error: '/home/brixy/project/bin/' is a directory

**Comment by medvednikov** [comment, marked as answer]
@Brixy please create a github issue. Will be implemented.

**Comment by Brixy** [comment]
Thanks a lot.

Done in #21766.
